lang/py/src/avro/schema.py

```diff
@@ -166,68 +170,62 @@ def to_json(self, names):
 class Name(object):
   """Class to describe Avro name."""
 
+  _base_name_pattern = re.compile(r'(?:^|\.)[A-Za-z_][A-Za-z0-9_]*$')
```

Comment:
Hmmm -- if python tightens up the name validation (and it should!), why not validate possible fullnames?  As this is, it permits invalid namespaces to pass through, like `schema.Name('123.a', None, None)` or `schema.Name('a', '123', None)`